server/src/main/java/com/thoughtworks/go/domain/JobAgentMetadata.java

```diff
@@ -25,21 +26,23 @@
 import java.util.Collection;
 import java.util.LinkedHashMap;
 import java.util.Map;
+import java.util.Objects;
 import java.util.stream.Collectors;
 
 public class JobAgentMetadata extends PersistentObject {
     private Long jobId;
     private String metadata;
```

Comment:
Since metadata is there with metatadataVersion we should make use of it - 

e.g.  `metadataVersion=2.0` and the content can be modeled as 

```json
{ 
    "profile" : {...},
    "cluster_profile": {...}
}
```

Now with the current implementation column name `metadata` does not make sense. It is better to use metadata or rename the column to `elasticprofilemetadata`

Reply:
how about `elasticAgentProfileMetadata`?